src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -91,6 +91,129 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException();
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0]))) //current drive rooted
+            {
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if(length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar && !PathInternal.IsDirectorySeparator(path[2]))
+            {
+                if (GetPathRoot(path) == GetPathRoot(basePath))
+                {
+                    return RemoveRelativeSegments(CombineNoChecks(basePath, path.Substring(2)), basePath.Length);
+                }
+                else
+                {
+                    return RemoveRelativeSegments(path.Insert(2, "\\"), PathInternal.GetRootLength(path));
+                } 
+            }
+
+            return GetFullPath(CombineNoChecks(basePath, path));
+        }
+
+        /// <summary>
+        /// Try to remove relative segments from the given path (without combining with a root).
+        /// </summary>
+        /// <param name="skip">Skip the specified number of characters before evaluating.</param>
+        private static string RemoveRelativeSegments(string path, int skip = 0)
```

Comment:
This looks like a exact copy of a method from the Unix-specific. You should move it to Path.cs that is shared between Windows and Unix instead.